src/amplitude.js

```diff
@@ -812,8 +814,44 @@ var _isNumber = function _isNumber(n) {
 };
 
 /**
- * Log revenue event with a price, quantity, and product identifier.
+ * Log revenue with Revenue interface. The new revenue interface allows for more revenue fields like
+ * revenueType and event properties.
  * @public
+ * @param {Revenue} revenue_obj - the revenue object containing the revenue data being logged.
+ * @example var revenue = new amplitude.Revenue().setProductId('productIdentifier').setPrice(10.99);
+ * amplitude.logRevenueV2(revenue);
+ */
+Amplitude.prototype.logRevenueV2 = function logRevenueV2(revenue_obj) {
+  if (!this._apiKeySet('logRevenueV2()')) {
+    return;
+  }
+
+  // if revenue input is a proxied object created by the async loading snippet, convert it into an revenue object
+  if (type(revenue_obj) === 'object' && revenue_obj.hasOwnProperty('_q')) {
+    var instance = new Revenue();
+    for (var i = 0; i < revenue_obj._q.length; i++) {
+        var fn = instance[revenue_obj._q[i][0]];
+        if (type(fn) === 'function') {
+          fn.apply(instance, revenue_obj._q[i].slice(1));
```

Comment:
Unfortunately not. It's because the Identify and Revenue proxy objects have to return this to mimic the real class behavior (they return this so you can chain function calls together like Revenue.setThis.setThat, etc). Whereas the amplitude functions don't return this. But that's a minor detail I guess...
I also just realized, any proxied getter methods on amplitude are broken since the proxied versions don't return anything. I guess that's why we tell people to get values using a callback to the init function, so that they are guaranteed the sdk is loaded.